src/transformStateWithClones.js

```diff
@@ -7,7 +7,56 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultedArray = [];
+  let currentObj = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
```

Comment:
Describe all variants of cases in constant.
const ACTION_CASES = {
     add: 'addProperties',
}